src/reynir_correct/errtokenizer.py

```diff
@@ -296,12 +296,33 @@ def word(cls, txt: str, val: Optional[BIN_TupleList] = None) -> "CorrectToken":
         return cls(TOK.WORD, txt, val)
 
     def __repr__(self) -> str:
-        return "<CorrectToken(kind: {0}, txt: '{1}', val: {2})>".format(
-            TOK.descr[self.kind], self.txt, self.val
+        return "<CorrectToken(kind: {0}, txt: '{1}', val: {2}, span: {3})>".format(
+            TOK.descr[self.kind], self.txt, self.val, self.origin_spans
         )
 
     __str__ = __repr__
 
+
+    def concatenate(self, other: "CorrectToken", *, separator: str = "", metadata_from_other: bool = False) -> "CorrectToken":
+        new_kind = self.kind if not metadata_from_other else other.kind
+        new_val = self.val if not metadata_from_other else other.val
+        self_txt = self.txt or ""
+        other_txt = other.txt or ""
+        new_txt = self_txt + separator + other_txt
+        self_original = self.original or ""
+        other_original = other.original or ""
+        new_original = self_original + other_original
+        new_cap = self._cap
+        new_err = self._err
+
+        self_origin_spans = self.origin_spans or []
+        other_origin_spans = other.origin_spans or []
+        separator_origin_spans: List[int] = ([len(self_original)] * len(separator) if len(other_origin_spans) > 0 else [])
```

Comment:
This probably doesn't work if len(separator) > 1 - then you need some kind of for loop over the separator characters, adding the indices (0, 1, ...) to the span entries. I'm not sure that separator is ever anything other than "" or " ", but if that is the assumption, it should be made explicit via asserts and comments.

Reply:
OK, now I understand that the separator is inserted even if it doesn't occur in the source text... which is maybe a questionable thing to do. Would be good to take a closer look at this and how it is actually used in practice.